File: ios/Stopwatch/platform/di/impl/LapsViewModelFactoryImpl.swift
class LapsViewModelFactoryImpl<SS, SN>: LapsViewModelFactory
where SS: MutableStateStore, SS.State == StopwatchState, SN: StackNavigator {
    private let stopwatchStore: SS
    private let loggingUseCase: LoggingUseCase
    private let startStopwatchUseCase: StartStopwatchUseCase
    private let newLapUseCase: NewLapUseCase
    private let stringTimeMapper: StringTimeMapper
    private let stackNavigator: SN
    
    init(
        _ stopwatchStore: SS,
        _ loggingUseCase: LoggingUseCase,
        _ startStopwatchUseCase: StartStopwatchUseCase,
        _ newLapUseCase: NewLapUseCase,
        _ stringTimeMapper: StringTimeMapper,
        _ stackNavigator: SN
    ) {
        self.stopwatchStore = stopwatchStore
        self.loggingUseCase = loggingUseCase
        self.startStopwatchUseCase = startStopwatchUseCase
        self.newLapUseCase = newLapUseCase
        self.stringTimeMapper = stringTimeMapper
        self.stackNavigator = stackNavigator
    }
    
    func make() -> some LapsViewModel {
        return LapsViewModelImpl(
            stopwatchStore,
            loggingUseCase,
            startStopwatchUseCase,
            newLapUseCase,
            stringTimeMapper,
            stackNavigator
        )
    }
}

File: android/app/src/main/java/com/rafaeltmbr/stopwatch/platform/di/impl/LapsViewModelFactoryImpl.kt
package com.rafaeltmbr.stopwatch.platform.di.impl

import androidx.compose.runtime.Composable
import androidx.lifecycle.viewmodel.compose.viewModel
import androidx.lifecycle.viewmodel.initializer
import androidx.lifecycle.viewmodel.viewModelFactory
import com.rafaeltmbr.stopwatch.core.data.stores.MutableStateStore
import com.rafaeltmbr.stopwatch.core.entities.StopwatchState
import com.rafaeltmbr.stopwatch.core.services.LoggingService
import com.rafaeltmbr.stopwatch.core.use_cases.NewLapUseCase
import com.rafaeltmbr.stopwatch.core.use_cases.StartStopwatchUseCase
import com.rafaeltmbr.stopwatch.platform.di.LapsViewModelFactory
import com.rafaeltmbr.stopwatch.platform.presentation.mappers.StringTimeMapper
import com.rafaeltmbr.stopwatch.platform.presentation.navigation.StackNavigator
import com.rafaeltmbr.stopwatch.platform.presentation.view_models.LapsViewModel
import com.rafaeltmbr.stopwatch.platform.presentation.view_models.impl.LapsViewModelImpl

class LapsViewModelFactoryImpl(
    private val stopwatchStore: MutableStateStore<StopwatchState>,
    private val stackNavigator: StackNavigator,
    private val startStopwatchUseCase: StartStopwatchUseCase,
    private val newLapUseCase: NewLapUseCase,
    private val loggingService: LoggingService,
    private val stringTimeMapper: StringTimeMapper
) : LapsViewModelFactory {
    @Composable
    override fun make(): LapsViewModel {
        val factory = viewModelFactory {
            initializer {
                LapsViewModelImpl(
                    stopwatchStore,
                    stackNavigator,
                    startStopwatchUseCase,
                    newLapUseCase,
                    loggingService,
                    stringTimeMapper,
                )
            }
        }

        return viewModel<LapsViewModelImpl>(factory = factory)
    }
}